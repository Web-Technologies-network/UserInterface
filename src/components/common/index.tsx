export { Card } from './Card/Card';
export { Danger } from './Typography/Danger';
export { Info } from './Typography/info';
export { Success } from './Typography/Success';
export { Warning } from './Typography/Warning';
