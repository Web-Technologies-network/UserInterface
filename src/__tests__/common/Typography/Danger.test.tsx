import React from 'react';
import { Danger } from '../../../components/common';
import { act } from 'react-dom/test-utils';
import ReactDOM from 'react-dom';
import testRenderer from 'react-test-renderer';

describe('Danger', () => {
  let container: string | Node | null; // this will hold the main div where our test component is rendered
  // before each test hook aka init
  beforeEach(() => {
    container = document.createElement('div');
    document.body.append(container);
  });
  // tear down
  afterEach(() => {
    document.body.removeChild(container as Node);
    container = null;
  });
  it('should render children', function () {
    act(() => {
      ReactDOM.render(
        <Danger>
          <div>children</div>
        </Danger>,
        container,
      );
      const grid = container.querySelector('div');
      expect(grid.textContent).toBe('children');
    });
  });
  it('should render the same', () => {
    const tree = testRenderer.create(
      <Danger>
        <div>test children</div>
      </Danger>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
