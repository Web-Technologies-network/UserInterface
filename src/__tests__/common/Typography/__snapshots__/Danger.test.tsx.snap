// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Danger should render the same 1`] = `
<div
  className="MuiGrid-root makeStyles-defaultFontStyle-1 makeStyles-dangerText-11"
>
  <div>
    test children
  </div>
</div>
`;
